compiler/src/iree/compiler/Codegen/LLVMCPU/KernelDispatch.cpp

```diff
@@ -1170,24 +1202,28 @@ setRootConfig(func::FuncOp entryPointFn,
   // smaller than `minTileSizes`, so we have to adjust the cache sizes again.
   cacheTileSizes = distTileSizes;
 
+  SmallVector<bool> distScalableTileFlags(distTileSizes.size(), false);
+  ScalableTileFlagsListType scalableTileFlags = {distScalableTileFlags,
+                                                 vecScalableFlags};
+
   LLVM_DEBUG(KD_DBGS() << "Distribution tile sizes: " << distTileSizes << "\n");
+  LLVM_DEBUG(KD_DBGS() << "Distribution scalable tile sizes: "
+                       << distScalableTileFlags << "\n");
   LLVM_DEBUG(KD_DBGS() << "Cache tile sizes: " << cacheTileSizes << "\n");
   LLVM_DEBUG(KD_DBGS() << "Vector tile sizes: " << vecTileSizes << "\n");
+  LLVM_DEBUG(KD_DBGS() << "Vector scalable tile sizes: " << vecTileSizes
+                       << "\n");
   LLVM_DEBUG(KD_DBGS() << "Vector scalable tile flags: " << vecScalableFlags
                        << "\n");
   LLVM_DEBUG(KD_DBGS() << "Vector size: " << vectorSize << "\n");
 
   if (usePeelingPipeline) {
-    // TODO: Use scalable vector sizes.
-    return setMatmulPeelingRootConfig(entryPointFn, contractionOp,
-                                      distTileSizes, cacheTileSizes,
-                                      vecTileSizes, vectorSize);
+    return setMatmulPeelingRootConfig(
+        entryPointFn, contractionOp, distTileSizes, cacheTileSizes,
+        vecScalableFlags, vecTileSizes, vectorSize);
```

Comment:
> Fundamentally, "scalability" is a concept only really available at the vector level, i.e. only the actual vector registers are scalable.

Based on the reply from other comments, we don't need to declare `distScalableTileFlags` and `ScalableTileFlagsListType`. We can directly pass `vecScalableFlags` to the function. This saves us a level of access (like l.805 -- `const SmallVectorImpl<bool> &vecScalableDims = inputScalableTileFlags.back();`).

Reply:
You are right and that's what I do for `setMatmulPeelingRootConfig`. However, there's also `setMatmulRootConfig` that takes both for now. I was planning to simplify that in a follow-up patch (trying to avoid making too many changes at once).

Reply:
In this case, maybe we can pass `inputScalableTileFlags.back()` with ArrayRef<bool> type to `setMatmulPeelingRootConfig`? It helps make the logic in `setMatmulPeelingRootConfig` easier, IMHO.

Reply:
Agreed.

Updated  - roughly as you suggested (sorry, wrote the code before saw your comment).